### ISSUE TITLE
Refactor fetch and storage for Orderbook

### ISSUE THREAD

**Issue by alvyn279** [Lercerss/graphelier#53]
We now want the current Orderbook state to be stored in the Redux store. This way, when a price level data changes, there is no need to re-render the entirety of price level components.

# Tasks
- Move away from data passing in `props`
- PriceLevel components should listen to changes in the store, that contains all the most up-to-date bids and asks for a price level.
- Highest level component should keep track of all the price levels to then map them to PriceLevel components

Highest level component will decide:
- if the response makes it so that the max quantities changes, re-render the whole orderbook
- if the response does not change the max quantity, spread changes to redux store, and only update a single price level component.

This will significantly improve the performance of the app, and will only re-render components that need to be re-rendered, in the context of moving message per message (#52)

[UPDATE]: To obtain better performance, one can just block the re-rendering at `PriceLevel` component with a `shouldComponentUpdate()`. This will avoid using Redux at all.

**Comment by alvyn279** [comment]
```
====== REFACTOR - FLOW FOR MESSAGE INCREMENTS =======


PriceLevelComponent: price as native props, orderBookEntry & maxQuantity as redux props/state
orderBookEntry: {
	price: 103.4,
	orders: [...],
	type: 'ask',
	isMiddle: 'false',
}

CASE new date/timestamp chosen => requests completely new orderbook (setState, re-render of the whole table)

CASE move messages by increments/decrements of X {

	received at least one non-existing price level: {
		append to orderBook listItems in store / modify orderBook list items price levels
		O(n) scan to update max quantity
		detect change in store listItems length in OrderBookSnapshot an setState from there
	}
	

	received EXISTING price level(s) ONLY: {
		update price level(s) in store listItems in store
		update maximum quantity (by doing an O(n) pass on orderBook listItems in the store)
		perform a re-render ONLY on the existing PriceLevelComponent
	}
}
```

**Comment by alvyn279** [comment]
From the discussion during the meeting, only we need `OrderBookSnapshot` component to be connected to Redux store and passing the information through props.

This allows for better maintainability and code clarity for `PriceLevel` component.

**Comment by alvyn279** [comment]
Solved in #84
